test/gtest/uct/test_sockaddr.cc

```diff
@@ -729,6 +742,43 @@ class test_uct_sockaddr : public uct_test {
         }
     }
 
+    static void conn_request_wait_listener_close_cb(
+            uct_listener_h listener, void *arg,
+            const uct_cm_listener_conn_request_args_t *conn_req_args)
+    {
+        test_uct_sockaddr *self = static_cast<test_uct_sockaddr*>(arg);
+
+        self->handle_conn_request(listener, conn_req_args,
+                                  test_uct_sockaddr::server_connect_cb,
+                                  test_uct_sockaddr::server_disconnect_cb);
+
+        ASSERT_EQ(nullptr, m_server_uct_listener);
+
+        // Notify listener_close_thread to destroy the listener
+        m_close_listener_notify = true;
+
+        // Wait until uct_listener_destroy is being called from
+        // listener_close_thread and dereference the server's listener to check
+        // whether the object is valid or not
+        while (m_server_uct_listener == nullptr) {
+            std::this_thread::yield();
+        }
+
+        ASSERT_TRUE(m_server_uct_listener.load()->cm != nullptr);
+
+        self->conn_req_recv_cnt_inc();
```

Comment:
why need to delay this call? I dont thisk this counter is critical for this test

Reply:
yes, you're right - we don't need to delay incrementing this counter.
so, returned to the way as it was